### ISSUE TITLE
Sections and paragraph names resolution

### ISSUE THREAD

**Issue by wiztigers** [TypeCobolTeam/TypeCobol#351]
**SPAWN OF #343**

Currently, paragraphs are not added to the symbol table.
To do it properly, I must know which paragraph names are in conflict in the following procedure division, and which ones are properly resolved by the IBM compiler.

```cobol
PROCEDURE DIVISION.

P1.
    CONTINUE.

P3.
    CONTINUE.

S1 SECTION.
P2.
    CONTINUE.
P3.
    CONTINUE.
P4.
    CONTINUE.

S2 SECTION.
P4.
    CONTINUE.
P5.
    CONTINUE.

P5 SECTION.
P6.
    CONTINUE.

S1 SECTION.
P7.
    CONTINUE.
```

Until further information, I'll assume the following:
* Declarations of `P1`, `P2`, `P5`, `P6`, `P7`, `S2` are OK
* Declarations of `P3`, `P4`, `S1` are ambiguous
* Uses of `P1`, `P2`, `P6`, `P7`, `S2` are OK
* Uses of `P3`, `P4`, `P5`, `S1` are ambiguous

In other words, I'll add every paragraph and section to the symbol table ; however I won't use a section containing a given paragraph to disambiguate this paragraph name.

If my assumption is false, I need the behaviour of some `PERFORM` and `PERFORM ... THROUGH` statements to help me implement how this kind of stuff works in COBOL.

**Comment by wiztigers** [comment]
@smedilol @delmasgu @reydelpa , thanks in advance for your input.

**Comment by wiztigers** [comment]
An exhaustive test should create a program using the procedure division that is in this issue's description:
* in the main program
* in a nested program
* in a stacked program

... all at the same time.

**Comment by smedilol** [comment]
```cobol
PROCEDURE DIVISION.
           perform P1
*ID:IGYPA3006-S Message:Procedure-name"P1"was multiply defined.  The statement was discarded.
           perform P2
           perform P3
           perform P4
*ID:IGYPA3006-S Message:Procedure-name"P4"was multiply defined.  The statement was discarded.
           perform P5
           perform P6
           perform P7
           perform S1
*ID:IGYPA3006-S Message:Procedure-name"S1"was multiply defined.  The statement was discarded.
           perform S2
           perform P51
      
           .
       P1.
           CONTINUE.
       P1.
           CONTINUE.
      
       P3.
           continue.
      
       S1 SECTION.
       P2.
           perform S1.
*ID:IGYPA3006-S Message:Procedure-name"S1"was multiply defined.  The statement was discarded.
       P3.
           CONTINUE.
       P4.
           CONTINUE.
      
       S2 SECTION.
       P4.
           perform P3.
*ID:IGYPA3006-S Message:Procedure-name"P3"was multiply defined.  The statement was discarded.
       P5.
           CONTINUE.
 *Renamed from `P5` to `P51`
       P51 SECTION.
       P6.
           CONTINUE.
      
       S1 SECTION.
       P7.
           CONTINUE.
```
After test:
* Declarations of `P5` is ambiguous in the first example (renamed `P51` in this example)
* All others declarations are ok

Rules
* You can declare 2 or more sections with the same name but you won't be able to `perform` them (example `S1`)
* You can declare 2 or more paragraphs with the same name. Whatever if paragraphs are in the same section or not. (example `P1`)
 * A section and a paragraph can't have the same name. Even if the paragraph is declared in another section (example of `P5` in the first comment of this issue)
* If a `perform` reference a paragraph within the same section, then only paragraph with this name must exists. If another paragraph with the same exists in another section, then it's no problem. (example `P3` just after the procedure division) 
*  If a `perform` reference a paragraph in another section and 2 or more sections contains the same paragrah name, then the call is ambiguous (example `P3` under `P4` paragraph) 
* If a `perform` reference a section, then the section must be uniquely defined (example `S1`)

**Comment by smedilol** [comment]
Need to create an unit test and see if these rules are correctly implemented.

**Comment by smedilol** [comment]
Fixed by #1710
